src/core/net/srp_client.cpp

```diff
@@ -1524,6 +1566,22 @@ void Client::HandleTimer(void)
         GrowRetryWaitInterval();
         SetState(kStateToUpdate);
         InvokeCallback(kErrorResponseTimeout);
+
+#if OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE && OPENTHREAD_CONFIG_SRP_CLIENT_SWITCH_SERVER_ON_FAILURE
+
+        // After certain number of back-to-back timeout failures, we try
+        // to switch the server. This is again done after invoking the
+        // callback. It works correctly due to the guard check at the
+        // top of `SelectNextServer()`.
+
+        mTimoutFailureCount++;
+
+        if (mTimoutFailureCount >= kMaxTimeoutFailuresToSwitchServer)
+        {
+            mTimoutFailureCount = 0;
```

Comment:
Seems we forget to reset `mTimoutFailureCount` in `Client::ProcessResponse`. Should `mTimoutFailureCount` be reset in `SelectNextServer` only when we really switched to a new server?

Reply:
> Seems we forget to reset mTimoutFailureCount in Client::ProcessResponse

I think we reset it already. It's around  line ~1158 in new code from this PR in `ProcessResponse()` method.

> Should `mTimoutFailureCount` be reset in `SelectNextServer()` only when we really switched to a new server?

Good suggestion. Thanks. I will change it in next push.